src/main/webapp/js/client.js

```diff
@@ -155,3 +155,111 @@ function getRestaurants() {
+      document.body.appendChild(para2);
+      var logoutLink = loginInfo[1];
+      var a2 = document.createElement('a');
+      a2.href = logoutLink;
+      a2.title = logoutLink;
+      a2.appendChild(document.createTextNode("here"));
+      var para3 = document.createElement("p");
+      para3.appendChild(document.createTextNode("Logout "));
+      para3.appendChild(a2);
+      document.body.appendChild(para3);
+    }
+  })
+} 
+
+function getRegister() {
+  console.log("at register page");
+  fetch('/login').then(response => response.json()).then((loginInfo) => {
+    if (loginInfo.length == 1) { // means user is not logged in
+      console.log("loginInfo.length == 1");
+      //var parent = document.getElementById("para");
+      var loginLink = loginInfo[0];
+      window.location.href = loginLink;
+    } if (loginInfo.length == 2) {
+      console.log("loginInfo.length == 2");
+      var userEmail = loginInfo[0];
+      var userInfo = document.getElementById("user-info");
+      var message = "You have registered as " + userEmail;
+      userInfo.appendChild(document.createTextNode(message));
+    } else { // three element means user already registered
+      console.log("loginInfo.length == 3");
+      var signInLink = "signin.html";
+      window.location.href = signInLink;
+    }
+  })
+}
+
+function controlButton() {
+    // remove sign in and register button if user is already signin;
+    fetch('/login').then(response => response.json()).then((loginInfo) => {
+        // loginInfo has different length base on user's state
+        // it has length of 1 if user is not logged in, 2 if user is logged
+        // in but not registered, and 3 if user is logged in and registered
+        var notLoggedIn = 1;
+        var loggedInAndRegistered = 3;
+        var inboxButton = document.getElementById("inbox-button");
+        var donateButton = document.getElementById("donation-button");
+        if (loginInfo.length == notLoggedIn) {
+            // user is not sign in
+            inboxButton.parentNode.removeChild(inboxButton);
+            donateButton.parentNode.removeChild(donateButton);
+            return;
+        }
+        var userEmail = loginInfo[0];
+        var registerButton = document.getElementById("register-button");
+        var signInButton = document.getElementById("sign-in-button");
+        //var userInfo = document.createElement("p");
+        //userInfo.appendChild(document.createTextNode(userEmail));
+        signInButton.parentNode.removeChild(signInButton);
+        //registerButton.parentNode.replaceChild(userInfo, registerButton);
+        //registerButton.innerHTML.replace(userEmail);
+        registerButton.innerHTML = userEmail;
+
+        if (loginInfo.length == loggedInAndRegistered) {
+            // the user-type is known
+            const restaurantType = "restaurant";
+            const charityType = "charity";
+            console.log(loginInfo[2]);
```

Comment:
remove logging from production as anyone viewing the website can see these details in their console.

Reply:
ok